server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -112,42 +135,80 @@ private void maybeScheduleNextRun(TimeValue time) {
+                inProgress.set(false);
+                return;
+            }
+            DiskHealthInfo previousHealth = this.lastReportedDiskHealthInfo.get();
             DiskHealthInfo currentHealth = diskCheck.getHealth(healthMetadata, clusterState);
             if (currentHealth.equals(previousHealth) == false) {
-                logger.debug("Health status changed from {} to {}", previousHealth, currentHealth);
-                this.lastReportedDiskHealthInfo = currentHealth;
+                String nodeId = clusterService.localNode().getId();
+                String healthNodeId = lastSeenHealthNode.get();
+                ActionListener<AcknowledgedResponse> listener = ActionListener.wrap(response -> {
+                    // Update the last reported value only if the health node hasn't changed.
+                    if (Objects.equals(healthNodeId, lastSeenHealthNode.get())
+                        && lastReportedDiskHealthInfo.compareAndSet(previousHealth, currentHealth)) {
+                        logger.debug(
+                            "Health info [{}] successfully send, last reported value: {}.",
+                            currentHealth,
+                            lastReportedDiskHealthInfo.get()
+                        );
+                    }
+                }, e -> logger.error(() -> format("Failed to send health info [%s] to health node, will try again.", currentHealth), e));
+                client.execute(
+                    UpdateHealthInfoCacheAction.INSTANCE,
+                    new UpdateHealthInfoCacheAction.Request(nodeId, currentHealth),
+                    ActionListener.runAfter(listener, () -> {
+                        inProgress.set(false);
+                        // Scheduling happens after the flag inProgress is false, this ensures that
+                        // if the feature is enabled after the following schedule statement, the setEnabled
+                        // method will be able to schedule the next run, and it will not be a no-op.
+                        // We prefer to err towards an extra scheduling than miss the enabling of this feature alltogether.
+                        maybeScheduleNextRun(monitorInterval);
+                    })
+                );
+            } else {
+                inProgress.set(false);
+                maybeScheduleNextRun(monitorInterval);
             }
-            inProgress.set(false);
-            // Scheduling happens after the flag inProgress is false, this ensures that
-            // if the feature is enabled after the following schedule statement, the setEnabled
-            // method will be able to schedule the next run, and it will not be a no-op.
-            // We prefer to err towards an extra scheduling than miss the enabling of this feature alltogether.
-            maybeScheduleNextRun(monitorInterval);
         }
     }
 
     DiskHealthInfo getLastReportedDiskHealthInfo() {
```

Comment:
I think we should mark this as `Nullable`

Reply:
Done.